storage/innobase/btr/btr0cur.cc

```diff
@@ -756,6 +756,7 @@ static bool btr_cur_need_opposite_intention(const buf_page_t &bpage,
                                             ulint compress_limit,
                                             const rec_t *rec)
 {
+  ut_ad(bpage.frame == page_align(rec));
   if (UNIV_LIKELY_NULL(bpage.zip.data) &&
```

Comment:
> What kind of testing are you looking for? Several mtr tests will fail when the debug assertion is added but the fix is omitted. The description mentions one of them: ./mtr innodb.temp_table_savepoint.
1. The bug is tagged as hang. Can we have some scenario which can produce a hang ?
2. I guess you mean the assert above, Let me check it out.

Reply:
Ok, The added assert fails. The block was misplaced and the removal looks correct. Looks like page_cur.rec is not in block->page at this point. Let me check bit more on this.

Reply:
I don’t think that the hang would be practical to reproduce. The closest that we have gotten to that is the assertion failure that was reported in [MDEV-31815](https://jira.mariadb.org/browse/MDEV-31815). Because I do not know if it is exactly due to this bug, I filed this bug separately. The purpose of that assertion is to catch potential hangs when the [WL#6326](https://dev.mysql.com/worklog/task/?id=6326) rules are being violated. That is, the index pages would be accessed in the wrong order while not holding an exclusive latch on the index tree.

Yes, the added assertion would fail quite easily if we omitted the fix from the caller.

Reply:
The removal is correct.